### ISSUE TITLE
Support for Android export

### ISSUE THREAD

**Issue by Harshharsh2275** [touilleMan/godot-python#251]
Just want to ask is there any possible way that we can export our Godot-python game for Android if so can you describe it step by step . Needed for a tutorial video 

Thanks for your efforts 👍

**Comment by touilleMan** [comment]
Hi @Harshharsh2275 

That's kind of funny you ask Android support given I'm currently working as my daily job (not related to Godot) on porting Python application on Android.

Long story short having Python running on Android is a big mess, having C extensions on top of that is a nightmare :'(
So I don't think there will be any Android support for the moment (of course if someone is passing by and wants to work on this, I'll be happy to answer questions ^^)

**Comment by dc740** [comment]
Hi, I'm passing by...
I was thinking (not even beginning) about using Kivy, or even pyqtdeploy, to try to export to Android. Since you already have experience on this, what's your take on those approaches, and in case you think one of them can be considered, where do you think it would be a good starting point.

Thanks!

**Comment by tavurth** [comment]
Interested in this also @touilleMan

**Comment by touilleMan** [comment]
@dc740 sorry for not answering sooner ;-)

I've tried to use pyqtdeploy for Android but it's almost unusable. pyqtdeploy is maintained by RiverBank computing, which is a one-person company from what I understood. So the project feel like very rough (few documentation, hard to access the source code (especially with a dependency such as Qt which is very big and complex to compile). I have the feeling pyqtdeploy is not usable outside of very straightforward usecase, or if you ask RiverBank for pay support.

Regarding Kivy, they have created the python4android project (p4a for short) to be able to run Pyhton and Android. This is really cool (I think it's the most advanced tool for porting Python application to Android), and it's the tool we used to port our application.
However p4a feels more like a hacky script written to solve the usecase of the Kivy team (i.e. run Kivy application on Android) than a real generic framework.
It's really hard to customize the boilerplate code or to register hooks to do some special work between p4a steps given plenty of things that should be configurable are just hardcoded.
For instance it's really hard to mix Java code and Python (e.g. if you want to have your GUI in Java and use Python only for a part of your application), if you create a [foreground service](https://developer.android.com/guide/components/services#Types-of-services) with python you cannot customize it button in the android task bare (the user typically expect to be able to click on this button to have the application related to the service popping on screen)...
That said p4a is basically the only tool that allows you to cross-compile binary dependency, so you don't have much choice if you have this kind of needs (if you don't, then you should have a look at [briefcase](https://briefcase.readthedocs.io/en/latest/) which has a much better architecture that allow simpler customization)

tl;dr: If you want to do Python on Android, just be prepare to waste an incredible amount of time on silly issues :cry:
